.travis.yml

```diff
@@ -1,126 +1,104 @@
-    if [[ "${TRAVIS_OS_NAME}" == "linux" && "${CXX%%+*}" == "clang" && -n "$(ls -A ${LLVM_INSTALL})" ]]; then
-    if   [[ "${CXX}" == "clang++-3.6" ]]; then LLVM_VERSION="3.6.2";
-    elif [[ "${CXX}" == "clang++-3.7" ]]; then LLVM_VERSION="3.7.1";
-    elif [[ "${CXX}" == "clang++-3.8" ]]; then LLVM_VERSION="3.8.1";
-    elif [[ "${CXX}" == "clang++-3.9" ]]; then LLVM_VERSION="3.9.1";
-    fi
-    LLVM_URL="http://llvm.org/releases/${LLVM_VERSION}/llvm-${LLVM_VERSION}.src.tar.xz"
-    LIBCXX_URL="http://llvm.org/releases/${LLVM_VERSION}/libcxx-${LLVM_VERSION}.src.tar.xz"
-    LIBCXXABI_URL="http://llvm.org/releases/${LLVM_VERSION}/libcxxabi-${LLVM_VERSION}.src.tar.xz"
-    mkdir -p llvm llvm/build llvm/projects/libcxx llvm/projects/libcxxabi
-    travis_retry wget -O - ${LLVM_URL} | tar --strip-components=1 -xJ -C llvm
-    travis_retry wget -O - ${LIBCXX_URL} | tar --strip-components=1 -xJ -C llvm/projects/libcxx
-    travis_retry wget -O - ${LIBCXXABI_URL} | tar --strip-components=1 -xJ -C llvm/projects/libcxxabi
-    (cd llvm/build && cmake .. -DCMAKE_INSTALL_PREFIX=${LLVM_INSTALL})
-    (cd llvm/build/projects/libcxx && make install -j2)
-    (cd llvm/build/projects/libcxxabi && make install -j2)
-    export CXXFLAGS="-isystem ${LLVM_INSTALL}/include/c++/v1"
-    export LDFLAGS="-L ${LLVM_INSTALL}/lib -l c++ -l c++abi"
-    export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${LLVM_INSTALL}/lib"
-    fi
+# ===================
+# STAGE: test (linux)
+# ===================
 
 before_script:
-
-  - mkdir ~/robot-code
-  - mkdir ~/robot-install
-
-  - export ROBOT_CODE=~/robot-code
-  - export ROBOT_INSTALL=~/robot-install
-
-  - cd $ROBOT_CODE
-  - wget https://github.com/google/googletest/archive/release-1.8.0.zip
-  - unzip release-1.8.0.zip
-  - cd googletest-release-1.8.0
-  - mkdir build
-  - cd build
-  - cmake ../ -DBUILD_GTEST=ON -DBUILD_SHARED_LIBS=ON
-  - sudo make install
-
-  - cd $ROBOT_CODE
-  - git clone https://github.com/eigenteam/eigen-git-mirror.git
-  - cd eigen-git-mirror
-  - mkdir build
-  - cd build
-  - cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ../
-  - sudo make install
-
-  - cd $ROBOT_CODE
-  - git clone --recursive https://github.com/oxfordcontrol/osqp.git
-  - cd osqp
-  - mkdir build
-  - cd build
-  - cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=$ROBOT_INSTALL -DUNITTESTS=OFF ../
-  - make install
-  - export osqp_DIR=$ROBOT_INSTALL
+  - docker pull ubuntu:$UBUNTU
 
 script:
-  - cd $TRAVIS_BUILD_DIR
-  - mkdir build
-  - cd build
-  - cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DBUILD_TESTING=ON ../
-  - make
-  - make test
+  - >-
+    docker run -it \
+      -v $TRAVIS_BUILD_DIR:$TRAVIS_BUILD_DIR \
+      -v $HOME/.ccache:$HOME/.ccache \
+      -w $TRAVIS_BUILD_DIR \
+      --env CC \
+      --env CXX \
+      --env TRAVIS_BUILD_DIR \
+      --env TRAVIS_BUILD_TYPE \
+      --env TRAVIS_CMAKE_GENERATOR \
+      --env VALGRIND_TESTS \
+      --env COMPILE_BINDINGS \
+      --env FULL_DEPS \
```

Comment:
I think this three env variables are neither defined in the environment nor used in the scripts running in the Docker container.